ThingIFSDK/ThingIFSDK/Gateway/Gateway.swift

```diff
@@ -35,3 +35,18 @@ public struct Gateway: TargetThing, Equatable {
         self.vendorThingID = vendorThingID
     }
 }
+
+extension Gateway: JsonDeserializable {
+
+    init(_ json: [String : Any]) throws {
```

Comment:
@satoshikumano I read 2 apple official documents:

  - [Initialization](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/Initialization.html#//apple_ref/doc/uid/TP40014097-CH18-ID203)
  - [Error Handling](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/ErrorHandling.html#//apple_ref/doc/uid/TP40014097-CH42-ID508)

`Error Handling` describes about throwing error from initializers but the document does not say anything it is preferable or not.

I think we can throw error from initializer because `Error Handling` explain it.

If you do not prefer it, I will implement factory method instead of `init`.